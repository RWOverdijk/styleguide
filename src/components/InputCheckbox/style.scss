/** @define InputCheckbox */

@import "../../css/variables";
@import "../../css/mixins";

.Van-InputCheckbox {
  position: relative;
  width: 20px;
  height: 20px;
  display: inline-block;

  &-input {
    position: absolute;
    top: 0;
    left: 0;
    width: 20px;
    height: 20px;
    margin: 0;
    opacity: 0;
    z-index: 10;

    & + .Van-InputCheckbox-styled {
      /* postcss-bem-linter: ignore */
      &::before {
        content: '';
        position: absolute;
        top: 0;
        left: 0;
        width: 20px;
        height: 20px;
        background-color: $color-white;
        border: 1px solid $color-grey-light;
      }
    }

    &:checked + .Van-InputCheckbox-styled {
      /* postcss-bem-linter: ignore */
      &::after {
        content: 'âœ”';
        position: absolute;
        top: 1px;
        left: 3px;
        color: $color-pink-primary;
        animation-duration: 0.2s;
        animation-name: scale;
      }
    }
  }
}
