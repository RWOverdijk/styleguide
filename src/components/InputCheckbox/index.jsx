import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './style.scss';

class InputCheckbox extends PureComponent {
  render() {
    const { className, ...props } = this.props;

    return (
      <span className={classnames('Van-InputCheckbox', className)}>
        <input type="checkbox" className="Van-InputCheckbox-input" {...props} />
        <span className="Van-InputCheckbox-styled" />
      </span>
    );
  }
}

InputCheckbox.defaultProps = {
  className: ''
};

InputCheckbox.propTypes = {
  className: PropTypes.string
};

export default InputCheckbox;
