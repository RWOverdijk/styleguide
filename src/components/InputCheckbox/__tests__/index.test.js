import React from 'react';
import renderer from 'react-test-renderer';
import InputCheckbox from '../';

describe("<InputCheckbox>", () => {
  it("renders the correct markup", () => {
    const tree = renderer.create(
      <InputCheckbox />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <InputCheckbox className="my-class" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <InputCheckbox value="on" data-id="0" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
