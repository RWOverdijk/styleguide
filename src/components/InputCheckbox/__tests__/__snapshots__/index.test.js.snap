// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputCheckbox> renders the correct markup 1`] = `
<span
  className="Van-InputCheckbox"
>
  <input
    className="Van-InputCheckbox-input"
    type="checkbox"
  />
  <span
    className="Van-InputCheckbox-styled"
  />
</span>
`;

exports[`<InputCheckbox> renders with specific attributes 1`] = `
<span
  className="Van-InputCheckbox"
>
  <input
    className="Van-InputCheckbox-input"
    data-id="0"
    type="checkbox"
    value="on"
  />
  <span
    className="Van-InputCheckbox-styled"
  />
</span>
`;

exports[`<InputCheckbox> renders with specific className 1`] = `
<span
  className="Van-InputCheckbox my-class"
>
  <input
    className="Van-InputCheckbox-input"
    type="checkbox"
  />
  <span
    className="Van-InputCheckbox-styled"
  />
</span>
`;
